src/thunderbots/software/ai/hl/stp/evaluation/pass.h

```diff
@@ -0,0 +1,26 @@
+/**
+ * Declaration for STP pass-related evaluation functions
+ */
+
+#include <optional>
+
+#include "ai/world/ball.h"
+#include "ai/world/field.h"
+#include "ai/world/robot.h"
+#include "geom/point.h"
+
+/**
+ * Finds the best place for the given robot to intercept the given ball
+ *
+ * @param ball The ball to intercept
+ * @param field The field on which we want the intercept to occur
+ * @param robot The robot that will hopefully intercept the ball
+ *
+ * @return A pair holding the best place that the robot can move to in order to intercept
+ *         the ball, and the duration into the future at which the pass would occur,
+ *         relative to the timestamp of the robot. If no possible intercept could be found
+ *         within the field bounds, returns std::nullopt
```

Comment:
I'm not fully convinced this function should care about the field bounds, I think it should be the caller's responsibility to handle points outside the field.

Although I guess we can't get rid of the `optional` entirely because technically if the ball is moving faster than the robot max speed, and its predicted speed does not decay, it could never be intercepted.

Reply:
after offline discussion, nothing being changed here